### DISCUSSION TITLE
What's the difference between "wants" & "requires"?

### DISCUSSION THREAD

**Discussion by jarmos-san** [wbthomason/packer.nvim#876] [General]
I stumble across quite a lot of dotfiles which uses the `wants` keyword over `require` when adding a dependency for a plugin. Following is an example:

```lua
-- truncated code

    use {
        'feline-nvim/feline.nvim',
        event = { 'VimEnter' },
        config = function()
            require 'conf.feline'
        end,
        -- Adding a dependency here with the "wants" keyword
        wants = { 'nvim-web-devicons' },
    }

    use {
        'lewis6991/gitsigns.nvim',
        event = { 'BufWinEnter', 'BufNewFile' },
        config = function()
            require('conf.gitsigns').config()
        end,
        -- Adding a dependency here with the "requires" keyword
        requires = { 'plenary.nvim' },
    }

-- truncated code
```
 I couldn't find any mention of the `wants` keywords anywhere in the documentations (or perhaps I missing out on something). So, I was wondering how does either `wants` or `requires` affect `packer`'s behaviour?

**Comment by nyngwang** [comment]
@Jarmos-san Did you find the answer?

**Reply by jarmos-san** [reply]
None yet, else I would've shared an answer here for sure! I was hoping someone else more aware of the source code would be willing to share some insights though.

**Reply by nyngwang** [reply]
@Jarmos-san Here is [a comment](https://github.com/wbthomason/packer.nvim/discussions/871#discussioncomment-4467583) where someone provides all things necessary to understand `wants`. I also left a summary [there](https://github.com/wbthomason/packer.nvim/pull/264#issuecomment-1361283448) based on what I have read from [the commit message](https://github.com/wbthomason/packer.nvim/pull/264/commits/bc121b95126dce9f83e5d275c46569da2542d028) of that PR adding `wants` as a key.

**In short**: we need to avoid the usage of `wants` since it's just a hotfix(and some people have reported that its behavior was incorrect in some edge cases), it will probably get deprecated/merged into `after` or something (or we even don't need to worry about the ordering after v2.)

**Reply by jarmos-san** [reply]
Got it! Can you share an answer here so that I can add it as the accepted answer?
